engine/access/rpc/backend/backend.go

```diff
@@ -76,6 +77,7 @@ type Backend struct {
 	collections       storage.Collections
 	executionReceipts storage.ExecutionReceipts
 	connFactory       connection.ConnectionFactory
+	resultCache       *badger.Cache[flow.Identifier, *access.TransactionResult]
```

Comment:
doesn't look like we need to save a reference here since it's only used in `backendTransactions`